src/scene/graph-node.js

```diff
@@ -646,6 +682,26 @@ class GraphNode extends EventHandler {
         return null;
     }
 
+    /**
+     * Get the first ancestor node found in the graph with the name.
+     *
+     * @param {string} name - The name of the graph.
+     * @returns {GraphNode|null} The first node to be found matching the supplied name. Returns
+     * null if no node is found.
+     */
+    findParentByName(name) {
+        let current = this;
+
+        while (current) {
+            if (current.name === name)
+                return current;
+
+            current = current._parent;
+        }
+
+        return null;
+    }
```

Comment:
By not reimplementing the iteration logic everywhere, you can save quite some lines. Do you see reasons to not just do this?

```suggestion
    findParentByName(name) {
        return this.findOneParent('name', name);
    }
```

Reply:
I was thinking about performance though it wouldn't make such a difference for `findParentByName`, a bit more if implementing `findByName` with `findOne('name', name)` but I guess both should be changed to respect PlayCanvas coding standards (and these functions aren't supposed to be called each frame anyway).

Reply:
Its nearly impossible to judge performance without measuring. I wouldn't be surprised if V8 optimizes the heck out of `GraphNode#findOneParent` as soon its called often enough to justify JIT hot patching which optimizes it to opcode machine code level performance.